src/mitol/openedx_user_retirement/management/commands/retire_one_learner.py

```diff
@@ -0,0 +1,288 @@
+            if config["all_states"].index(start_state) < learner_state_index:
+                LOG("State {} completed in previous run, skipping".format(start_state))
+                continue
+
+            LOG("Starting state {}".format(start_state))
+
+            config["LMS"].update_learner_retirement_state(
+                username, start_state, "Starting: {}".format(start_state)
+            )
+
+            # This does the actual API call
+            start_time = time()
+            response = getattr(config[service], method)(learner)
+            end_time = time()
+
+            LOG(
+                "State {} completed in {} seconds".format(
+                    start_state, end_time - start_time
+                )
+            )
+
+            config["LMS"].update_learner_retirement_state(
+                username,
+                end_state,
+                "Ending: {} with response:\n{}".format(end_state, response),
+            )
+
+            learner_state_index += 1
+
+            LOG("Progressing to state {}".format(end_state))
+
+        config["LMS"].update_learner_retirement_state(
+            username, COMPLETE_STATE, "Learner retirement complete."
+        )
+        LOG("Retirement complete for learner {}".format(username))
+    except Exception as exc:  # pylint: disable=broad-except
+        exc_msg = _get_error_str_from_exception(exc)
+
+        try:
+            LOG("Error in retirement state {}: {}".format(start_state, exc_msg))
+            config["LMS"].update_learner_retirement_state(
+                username, ERROR_STATE, exc_msg
+            )
+        except Exception as update_exc:  # pylint: disable=broad-except
+            LOG(
+                "Critical error attempting to change learner state to ERRORED: {}".format(
+                    update_exc
+                )
+            )
+
+        FAIL_EXCEPTION(
+            ERR_WHILE_RETIRING,
+            'Error encountered in state "{}"'.format(start_state),
+            exc,
+        )
+
+
+class Command(BaseCommand):
+    """
+    Retire user from Open edX.
+    """
+
+    help = """
+    Retire users on edx. Username should be used to identify a user.
+
+    (OPTIONAL): You can also add `--initiate_retirement` flag to move user to the retirement
+    pipeline. If a user is already moved to the retirement pipeline in edX then DO NOT add this flag.
+    This helps bypass the need to run `retire_user` management command in edX and moves the user to Pending retirement
+    state.
+
+    You can use one of the following:
+    `./manage.py retire_users --user=foo`
+    `./manage.py retire_users -u foo`
+    """
+
+    def add_arguments(self, parser):
+        """parse arguments"""
+
+        # pylint: disable=expression-not-assigned
+        parser.add_argument(
+            "-u",
+            "--user",
+            action="append",
+            default=[],
+            dest="users",
+            help="Single or multiple username(s)",
```

Comment:
Tried to follow what we have for the current retirement command in MITxOnline. Retires the users list one by one in a loop. So, you can do `--user=test --user=test2`. It will retire both users one by one.